api/api/views/media_views.py

```diff
@@ -265,6 +294,8 @@ async def get_image_proxy_media_info(self) -> image_proxy.MediaInfo:
         ],
     )
 
+    # Thumbnails can always be cached
+    @async_method_cache_control(max_age=ttl(weeks=52), public=True)
```

Comment:
What would happen if the thumbnail response is flaky, or returns an error only on the first request? Would the error be cached, or are errors never cached?

Reply:
That is an interesting question. I need to look deeper into this in Django and make sure I understand everything correctly before saying definitely.. but I _think_ that because an error response should be raised, it would break the view function's routine before the cache control header is set. So when we raise the upstream thumbnail failure, that will stop the rest of the view from happening and hand off to the error handler middleware instead, which should prevent the cache control headers from being set because the cache_control decorator sets them on the response after the view function is finished.

But, again, I need to confirm this by trying it out locally and seeing what the behaviour is.

If it does not behave that way, we have two options:

1. We can make write a wrapper to the `patch_cache_control_headers` function to only patch onto status codes in a range we've defined, either per route or following a global setting for the app.
2. We can close the cache control issue as won't do, and instead rely on heuristic caching in browsers and define only edge TTLs in Cloudflare for specific routes.

These are the status codes that are described as cacheable by default when cache control headers are not defined (e.g., if we don't do this PR): https://datatracker.ietf.org/doc/html/rfc7231#section-6.1

We could, instead of setting the cache-control header, set the last-modified header. Then browsers can use heuristic caching as described [here in the RFC for cache controls](https://datatracker.ietf.org/doc/html/rfc7234#section-4.2.2). We would define edge caching behaviour in Cloudflare _or_ we can set _both_ and instruct Cloudflare to use the cache control header for edge TTL and re-write response headers to exclude the cache control header sent to browsers. So then Cloudflare uses the cache-control for edge caching instructions (with limitations on the range we allow) and all responses have headers re-written to exclude cache-control when forwarded to the requester.

That blends the two approaches and allows us to control caching entirely from the application code, without having to touch Cloudflare cache rules for the API. The frontend really needs different caching heuristics altogether, like maybe only an hour or caching, because UI changes are independent of the underlying data changes possible from the API.

I need to confirm two things though:

1. Do the cache-control headers get set for error responses; in other words, does raising the error response cause the cache control decorator to be bypassed and never run. And do we handle errors consistently enough to rely on this (i.e., do we _raise_ rather than _return_ error responses).
2. Does Cloudflare's browser TTL bypass setting allow completely leaving off the cache-control header or does it just re-write it to match edge TTL?

Our default browser TTL is 4 hours, which you can see on a request like https://api.openverse.engineering/v1/, and is configured entirely by Cloudflare.